core/blockchain_arbitrum.go

```diff
@@ -54,3 +67,52 @@ func (bc *BlockChain) ClipToPostNitroGenesis(blockNum rpc.BlockNumber) (rpc.Bloc
 	}
 	return blockNum, currentBlock
 }
+
+// finds the number of blocks that aren't prunable
+func (bc *BlockChain) FindRetentionBound() uint64 {
```

Comment:
If `TrieRetention` is 0, the behavior should be the same as standard geth, right? Might be good to have an early exit to make that clear